scripts/edit-task.js

```diff
@@ -0,0 +1,231 @@
+import { CLICK_EVENT, HIDE_CLASS } from "/scripts/constants.js";
+import createButton from "/scripts/factory/createButton.js";
+import toggleButtons from "/scripts/factory/toggleButton.js";
+
+function saveEvent(
+  doneButton,
+  editButton,
+  deleteButton,
+  saveAndDoneButton,
+  revertButton,
+  saveButton,
+  inputElement,
+  tasks,
+  id,
+  saveEventHandler,
+  saveAndDoneEventHandler,
+  revertEventHandler
+) {
+  inputElement.contentEditable = false;
+  tasks[id].content = inputElement.textContent;
+  toggleButtons(
+    editButton,
+    deleteButton,
+    revertButton,
+    saveButton,
+    doneButton,
+    saveAndDoneButton
+  );
+  console.log("call");
+  saveButton.removeEventListener(CLICK_EVENT, saveEventHandler);
+  revertButton.removeEventListener(CLICK_EVENT, revertEventHandler);
+  saveAndDoneButton.removeEventListener(CLICK_EVENT, saveAndDoneEventHandler);
+}
+
+function onTodoDone(
+  doneButton,
+  editButton,
+  deleteButton,
+  saveAndDoneButton,
+  revertButton,
+  saveButton,
+  inputElement,
+  toolbar,
+  tasks,
+  id,
+  editEventHandler,
+  saveEventHandler,
+  saveAndDoneEventHandler,
+  revertEventHandler
+) {
+  inputElement.contentEditable = false;
```

Comment:
duplicate [statement](https://github.com/rsa-wtag/Todo-App-with-JS/blob/dbb03950f212a1176151f9ac9fdc9c9f4dc9646f/scripts/edit-task.js#L19), consider introducing a reusable function.

```
function disableEditing(element, disabled = false) {
    element.contentEditable = disabled;
}
```

```suggestion
  disableEditing(inputElement);
```